### ISSUE TITLE
Callbacks & Polling

### ISSUE THREAD

**Issue by Laziemo** [yagop/node-telegram-bot-api#586]
Hello )) I am a junior developer and just starting to wrap my head around callbacks. I have made a few simple and fully functioning bots with node-telegram-bot-api and it has always been great to use! Big thanks to the team behind this library!

To improve my code and learn more on callbacks, I re-organized my code to turn processes into individual functions as much as possible and upon doing so I now get polling errors for any .onText function that makes use of callbacks. 

Here is an example of one function in the codebase that works and one that does not, including all the boiler-plate.

***Code Snippet***
//--------------------------------------------------------------
"use strict"
//--------------------------------------------------------------
var TelegramBot = require('node-telegram-bot-api');
var request = require('request');
var express = require('express');
var cronJob = require('node-cron');
//--------------------------------------------------------------
var app = express();
var port = process.env.PORT || 13337;
var chatId = 0;
//--------------------------------------------------------------
app.listen(port);
app.get('/', function(req,res){
	res.status(200).send("CryptopiaBot is Live ...");
});
//--------------------------------------------------------------
const token = "ze'token";
const CryptopiaBot = new TelegramBot(token, {polling: true});
const marketAPI = 'https://www.cryptopia.co.nz/api/GetMarket/';
const ordersAPI = ' https://www.cryptopia.co.nz/api/GetMarketOrders/';
//---------------------------THIS FUNCTION WORKS FINE-----------------------------------
CryptopiaBot.onText(/help/, (msg,match)=>{
	chatId = msg.chat.id;
	CryptopiaBot.sendMessage(chatId,`***CryptopiaTrackerBot***\n CryptopiaTrackerBot tracks trades on Cryptopia Exchange and sends alerts.\nCommands: List all commands\n`);
});
//--------------------------THIS GIVES POLLING ERROR------------------------
CryptopiaBot.onText(/market (.+)/,requestMarket, (msg,match) =>{
	chatId = msg.chat.id;
	const coin = match[1];
		if(typeof requestMarket === 'function'){
			const marketData = requestMarket(coin);
				if(marketData!=0){
						CryptopiaBot.sendMessage(chatId, `${coin} Status:\nAsk Price: ${marketData.AskPrice} \nBid Price: ${marketData.BidPrice}\nChange: ${marketData.Change} \nOpen: ${marketData.Open}\nClose: ${marketData.Close} \nVolume: ${marketData.Volume}\nBuy Volume: ${marketData.BuyVolume}\nSell Volume: ${marketData.SellVolume}\n`);
				}
				else{
					CryptopiaBot.sendMessage(chatId, "Incorrect use of command: market\n");
					CryptopiaBot.sendMessage(chatId, "Ensure: \nCorrect ticker symbol for coin. *Upper Case*");
				}
		}
});
//--------------------------------------------------------------
var requestMarket = function(coin){
	const API = marketAPI.concat(coin+"_BTC");
	request(API, (error, response, body) =>{
		const marketInfo = JSON.parse(body);
			if(marketInfo.Data!=null&&marketInfo.Data!= undefined)
				return marketInfo.Data;
			
			else
				return 0;
	});
}
//--------------------------------------------------------------

All forms of feedback are welcome ))
Also, if someone could elaborate on what exactly polling is; ELI5, karma points for helping a keen fellow developer. 

Cheers,
Laziemo.

**Comment by jason-c-child** [comment]
🤷‍♂️ maybe not the best forum for this request; it bloats the issue count and should really be somewhere like stackoverflow 😄 the reason why im looking at this issue: i was evaluating this package as i may use it...some people look at a project with only a couple of contributors and a bunch of open issues and just move on...i've done that myself before

**Comment by Laziemo** [comment]
yeah figured :P cheers for the response. 

The issue here was that certain keys from the the JSON object being called at the API were returning as undefined. 
CryptopiaBot.sendMessage(chatId, **${coin}** Status:\nAsk Price: **${marketData.AskPrice}** \nBid Price: **${marketData.BidPrice}**\nChange: **${marketData.Change}** \nOpen: **${marketData.Open}**............
}

The preceeding if statement requires a check like if marketData!=null || marketData!=undefined
